packages/core-data/src/selectors.ts

```diff
@@ -365,7 +365,7 @@ export const getEntityRecord = createSelector(
 		}
 		const context = query?.context ?? 'default';
 
-		if ( query === undefined ) {
+		if ( ! query || ! query._fields ) {
```

Comment:
The new condition is the same as that used in the reducer.

https://github.com/WordPress/gutenberg/blob/2e69c4a1f175269004327759448097ef3a2336c8/packages/core-data/src/queried-data/reducer.js#L153-L161